### PR TITLE
Ian's fix for Cluster Docs

### PR DESCRIPTION
**Pull request by ijrsvt**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/cluster/config.rst [hunk 1]
```diff
@@ -493,6 +501,35 @@ The extra options to pass to ``docker run``.
...
+
+.. _cluster-configuration-disable-automatic-runtime-detection:
+
+``docker.disable_automatic_runtime_detection``
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If enabled, Ray will not try to use the NVIDIA runtime if GPUs are present.
+
+* **Required:** No
+* **Importance:** Medium
```
Comment (AmeerHajAli) [inline doc/source/cluster/config.rst:513]: should this be low importance?

File: doc/source/cluster/config.rst [hunk 2]
```diff
@@ -493,6 +501,35 @@ The extra options to pass to ``docker run``.
...
+* **Type:** Boolean
+* **Default:** ``False``
+* **Update requires:** Restart
```
Comment (AmeerHajAli) [inline doc/source/cluster/config.rst:516]: please remove this.

File: doc/source/cluster/config.rst [hunk 3]
```diff
@@ -493,6 +501,35 @@ The extra options to pass to ``docker run``.
 * **Default:** ``[]``
 * **Update requires:** Restart
 
+
+.. _cluster-configuration-disable-automatic-runtime-detection:
+
+``docker.disable_automatic_runtime_detection``
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If enabled, Ray will not try to use the NVIDIA runtime if GPUs are present.
```
Comment (AmeerHajAli) [inline doc/source/cluster/config.rst:510]:
```suggestion
If enabled, Ray will not try to use the NVIDIA Container Runtime if GPUs are present.
```

File: doc/source/cluster/config.rst [hunk 4]
```diff
@@ -493,6 +501,35 @@ The extra options to pass to ``docker run``.
...
+### disable_shm_size_detection
+.. _cluster-configuration-disable-shm-size-detection:
+
+``docker.disable_shm_size_detection``
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If enabled, Ray will not automatically specify the size /dev/shm for the started container.
+
+* **Required:** No
+* **Importance:** Medium
```
Comment (AmeerHajAli) [inline doc/source/cluster/config.rst:528]: low importance?

File: doc/source/cluster/config.rst [hunk 5]
```diff
@@ -493,6 +501,35 @@ The extra options to pass to ``docker run``.
...
+* **Type:** Boolean
+* **Default:** ``False``
+* **Update requires:** Restart
```
Comment (AmeerHajAli) [inline doc/source/cluster/config.rst:531]: please remove this.

File: doc/source/cluster/config.rst [hunk 6]
```diff
@@ -493,6 +501,35 @@ The extra options to pass to ``docker run``.
...
+
+
+### disable_shm_size_detection
+.. _cluster-configuration-disable-shm-size-detection:
+
+``docker.disable_shm_size_detection``
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If enabled, Ray will not automatically specify the size /dev/shm for the started container.
```
Comment (AmeerHajAli) [inline doc/source/cluster/config.rst:525]: what happens when ray does not automatically specifies it? what the default value will be?

File: doc/source/cluster/config.rst (hunk 1 above)
Comment (ijrsvt) [inline doc/source/cluster/config.rst:513]:
```suggestion
* **Importance:** Low
```

Comment (ijrsvt) [reply doc/source/cluster/config.rst:513]: yep

File: doc/source/cluster/config.rst (hunk 5 above)
Comment (ijrsvt) [inline doc/source/cluster/config.rst:531]:
```suggestion
* **Importance:** Low
* **Type:** Boolean
* **Default:** ``False``
```

File: doc/source/cluster/config.rst (hunk 2 above)
Comment (ijrsvt) [inline doc/source/cluster/config.rst:516]:
```suggestion
```

File: doc/source/cluster/config.rst [hunk 7]
```diff
@@ -493,6 +501,33 @@ The extra options to pass to ``docker run``.
...
+* **Default:** ``False``
+
+
+### disable_shm_size_detection
+.. _cluster-configuration-disable-shm-size-detection:
+
+``docker.disable_shm_size_detection``
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If enabled, Ray will not automatically specify the size /dev/shm for the started container.
```
Comment (ijrsvt) [inline doc/source/cluster/config.rst:524]:
```suggestion
If enabled, Ray will not automatically specify the size /dev/shm for the started container and the runtime's default value (64MiB for Docker) will be used.
```

File: doc/source/cluster/config.rst (hunk 6 above)
Comment (ijrsvt) [reply doc/source/cluster/config.rst:524]: See my suggestion below

File: doc/source/cluster/config.rst [hunk 8]
```diff
@@ -442,7 +450,7 @@ After you have customized the nodes, create a new machine image (or docker conta
 
 The Docker image to pull in the head and worker nodes. If no image is specified, Ray will not use Docker.
```
Comment (ijrsvt) [inline doc/source/cluster/config.rst:451]:
```suggestion
The default Docker image to pull in the head and worker nodes. This is overridden by the `head_image` and `worker_image` fields. If neither `image` nor (`head_image` and `worker_image`) are specified, Ray will not use Docker.
```

File: doc/source/cluster/config.rst [hunk 9]
```diff
@@ -53,6 +53,14 @@ Docker
     :ref:`pull_before_run <cluster-configuration-pull-before-run>`: bool
     :ref:`run_options <cluster-configuration-run-options>`:
         - str
+    :ref:`disable_automatic_runtime_detection <cluster-configuration-disable-automatic-runtime-detection>`: bool
+    :ref:`disable_shm_size_detection <cluster-configuration-disable-shm-size-detection>`: bool
+    :ref:`head_image <cluster-configuration-head-image>`: str
+    :ref:`worker_image <cluster-configuration-worker-image>`: str
```
Comment (AmeerHajAli) [inline doc/source/cluster/config.rst:59]: you added the links without the content:
```
    :ref:`head_image <cluster-configuration-head-image>`: str
    :ref:`worker_image <cluster-configuration-worker-image>`: str
```

**Comment by AmeerHajAli** [comment]
Incorporated the changes from this! Thank you very much, it was very helpful @ijrsvt !

**Pull request closed without merging**